### PR TITLE
Add ListStepPlot refactor _ListPlot a little

### PR DESCRIPTION
**Pull request by rocky**
* Add `ListStepPlot[]` 
* Refactor `_ListPlot[]` a little; note this is an abstract class. Add a ListPlot type enum
* mathics.eval.plot -> mathics.eval.drawing.plot

### CODE REVIEW

**Comment by rocky** [comment]
As always, this is not perfect; not all options are supported. But it is a starting point.

File: mathics/builtin/drawing/plot.py [hunk 1]
```diff
@@ -2067,14 +2067,16 @@ class ListStepPlot(_ListPlot):
     >> ListStepPlot[{1, 1, 2, 3, 5, 8, 13, 21}]
      = -Graphics-
 
-
-    By default ListStepPlots are joined, but that can be disabled.
+    'ListPlot' accepts a superset of the Graphics options. \
```
Comment (mmatera) [inline mathics/builtin/drawing/plot.py:2070]: Shouldn't be 'ListStepPlot'?

**Review by mmatera — APPROVED** [review]
LGTM

File: mathics/builtin/drawing/plot.py (hunk 1 above)
Comment (rocky) [reply mathics/builtin/drawing/plot.py]: Thanks - corrected now.

**Pull request merged**